masschallenge/impact-api
# [AC-3838] - Move sorl to use S3

**Changes Introduced in [AC-3838](https://masschallenge.atlassian.net/browse/AC-3838):**
- Set an S3 Storage Backend (using boto3 and django-storages) as the project's default storage in deployed settings.
- Explicitly set static files to rely on the file system storage.
- Serve media files URLs when in DEBUG mode.

**Test:**
- git checkout AC-3838-test
- Setup locally by editing your `.dev.env` file to include:
  ```
  AWS_STORAGE_BUCKET_NAME=accelerate-assets-testing
  AWS_ACCESS_KEY_ID=xxx
  AWS_SECRET_ACCESS_KEY=xxx
  ```
- replace AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY's values with either your own user's credentials, or the ones used by [Test1](https://console.aws.amazon.com/opsworks/home?region=us-east-1#/stack/adcaaee8-8beb-4d13-8ca0-edc2f558ab4d/show).
- run `make clean;make build;make dev; and `make runserver` in a separate window.
- `make load-remote-db DB_FILE_NAME=initial_schema.sql.gz`
- Login as a superuser (e.g., demoadmin@masschallenge.org)
- Go to http://localhost:8000/api/impact/startup/736/
  - Use the browsable API to PUT an image in the startup logo.
  - See that the file was uploaded, and that the displayed URL is accessible.

**Notes:**
- This Issue included a bunch of AWS configuration changes. They are documented in standards: https://github.com/masschallenge/standards/pull/55
- It is possible to now remove the `HAS_SORL` statements. Since they are already removed in `modular-models-epic`, I prefer to avoid this laborious duplicated work.
